src/Rules/Install.hs

```diff
@@ -0,0 +1,299 @@
+           version <- interpretInContext context (getPkgData Version)
+           -- Copy over packages
+           let targetDest = dest ++ ghclibDir -/- pkgNameString pkg ++ "-" ++ version
+           strip <- stripCmd context
+           ways <- interpretInContext context getLibraryWays
+           let ghcCabalInplace = inplaceBinPath -/- "ghc-cabal" -- HACK?
+
+           -- HACK (#318): copy stuff back to the place favored by ghc-cabal
+           quietly $ copyDirectoryContents (Not excluded)
+                       installDistDir (installDistDir -/- "build")
+           pref <- setting InstallPrefix
+           unit $ cmd ghcCabalInplace
+                      [ "copy"
+                      , pkgPath
+                      , installDistDir
+                      , strip
+                      , dest
+                      , pref
+                      , ghclibDir
+                      , docDir -/- "html/libraries"
+                      , intercalate "  " (map show ways) ]
+
+   -- Register packages
+  let installedGhcPkgReal = dest ++ binDir -/- "ghc-pkg" <.> exe
+  let installedGhcReal = dest ++ binDir -/- "ghc" <.> exe
+  unit $ cmd installedGhcPkgReal
+             [ "--force", "--global-package-db"
+             , installedPackageConf, "update"
+             , pkgConfInstallPath ] -- XXX: Extend builder to support installed bin
+  forM_ installPkgs $ \pkg@Package{..} -> do
+    when (isLibrary pkg) $
+        withLatestBuildStage pkg $ \stg -> do
+            let context = vanillaContext stg pkg
+            top <- interpretInContext context getTopDirectory
+            let installDistDir = top -/- buildPath context
+            let ghcCabalInplace = inplaceBinPath -/- "ghc-cabal" -- HACK?
+            ghclibDir <- installGhcLibDir
+            pref <- setting InstallPrefix
+            docDir <- installDocDir
+            r <- relocatableBuild
+            unit $ cmd ghcCabalInplace
+                       [ "register"
+                       , pkgPath
+                       , installDistDir
+                       , installedGhcReal
+                       , installedGhcPkgReal
+                       , dest ++ topdir
+                       , dest
+                       , dest ++ pref
+                       , dest ++ ghclibDir
+                       , dest ++ docDir -/- "html/libraries"
+                       , if r then "YES" else "NO" ]
+
+  confs <- getDirectoryContents installedPackageConf
+  forM_ confs (\f -> createData $ installedPackageConf -/- f)
+  unit $ cmd installedGhcPkgReal
+             [ "--force", "--global-package-db"
+             , installedPackageConf, "recache" ]
+  where
+    createData f = unit $ cmd ("chmod" :: String) [ "644", f ]
+    excluded = Or
+        [ Test "//haddock-prologue.txt"
+        , Test "//package-data.mk"
+        , Test "//setup-config"
+        , Test "//inplace-pkg-config"
+        , Test "//build" ]
+
+-- XXX (izgzhen): Do we need INSTALL_OPTS?
+-- installOpts = undefined
```

Comment:
Make scripts used `INSTALL_OPTS` a lot, however querying the `make show` shows that it is just `""` in my reference build, and I can't find any `*.mk` defining it either.

So I don't know if I should just ignore it.